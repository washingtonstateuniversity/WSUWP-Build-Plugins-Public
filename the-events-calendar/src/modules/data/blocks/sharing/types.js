/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/SHARING_BLOCK/SET_INITIAL_STATE`;

export const SET_GOOGLE_CALENDAR_LABEL = `${ PREFIX_EVENTS_STORE }/SET_GOOGLE_CALENDAR_LABEL`;

export const SET_ICAL_LABEL = `${ PREFIX_EVENTS_STORE }/SET_ICAL_LABEL`;

export const SET_HAS_ICAL = `${ PREFIX_EVENTS_STORE }/SET_HAS_ICAL`;
export const TOGGLE_ICAL = `${ PREFIX_EVENTS_STORE }/TOGGLE_ICAL`;

export const SET_HAS_GOOGLE_CALENDAR = `${ PREFIX_EVENTS_STORE }/SET_HAS_GOOGLE_CALENDAR`;
export const TOGGLE_GOOGLE_CALENDAR = `${ PREFIX_EVENTS_STORE }/TOGGLE_GOOGLE_CALENDAR`;
