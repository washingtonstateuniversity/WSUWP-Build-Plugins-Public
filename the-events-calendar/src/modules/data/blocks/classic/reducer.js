/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
