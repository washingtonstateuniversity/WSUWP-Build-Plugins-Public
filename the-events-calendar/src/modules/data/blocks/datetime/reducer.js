/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
