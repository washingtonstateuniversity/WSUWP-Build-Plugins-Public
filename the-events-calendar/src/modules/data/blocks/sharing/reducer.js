/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	googleCalendarLabel: __( 'Google Calendar', 'the-events-calendar' ),
	iCalLabel: __( 'iCal Export', 'the-events-calendar' ),
	hasiCal: true,
	hasGoogleCalendar: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_GOOGLE_CALENDAR_LABEL:
			return {
				...state,
				googleCalendarLabel: action.payload.label,
			};
		case types.SET_ICAL_LABEL:
			return {
				...state,
				iCalLabel: action.payload.label,
			};
		case types.SET_HAS_GOOGLE_CALENDAR:
			return {
				...state,
				hasGoogleCalendar: action.payload.hasGoogleCalendar,
			};
		case types.TOGGLE_GOOGLE_CALENDAR:
			return {
				...state,
				hasGoogleCalendar: ! state.hasGoogleCalendar,
			};
		case types.SET_HAS_ICAL:
			return {
				...state,
				hasiCal: action.payload.hasiCal,
			};
		case types.TOGGLE_ICAL:
			return {
				...state,
				hasiCal: ! state.hasiCal,
			};
		default:
			return state;
	}
};
