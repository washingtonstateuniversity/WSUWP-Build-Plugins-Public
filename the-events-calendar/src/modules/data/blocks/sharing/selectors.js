/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const sharingSelector = ( state ) => state.events.blocks.sharing;

export const googleCalendarLabelSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.googleCalendarLabel,
);

export const hasGooglecalendarLabel = createSelector(
	[ sharingSelector ],
	( values ) => values.hasGoogleCalendar,
);

export const iCalLabelSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.iCalLabel,
);

export const hasIcalSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.hasiCal,
);
