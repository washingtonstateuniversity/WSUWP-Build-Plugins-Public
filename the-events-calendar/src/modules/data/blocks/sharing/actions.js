/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';

export const setGoogleCalendarLabel = ( label ) => ( {
	type: types.SET_GOOGLE_CALENDAR_LABEL,
	payload: {
		label,
	},
} );

export const setHasGoogleCalendar = ( hasGoogleCalendar ) => ( {
	type: types.SET_HAS_GOOGLE_CALENDAR,
	payload: {
		hasGoogleCalendar,
	},
} );

export const toggleGoogleCalendar = () => ( { type: types.TOGGLE_GOOGLE_CALENDAR } );

export const setiCalLabel = ( label ) => ( {
	type: types.SET_ICAL_LABEL,
	payload: {
		label,
	},
} );

export const setHasIcal = ( hasiCal ) => ( {
	type: types.SET_HAS_ICAL,
	payload: {
		hasiCal,
	},
} );

export const toggleIcalLabel = () => ( { type: types.TOGGLE_ICAL } );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );
