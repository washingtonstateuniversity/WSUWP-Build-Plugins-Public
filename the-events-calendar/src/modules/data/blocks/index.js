/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
