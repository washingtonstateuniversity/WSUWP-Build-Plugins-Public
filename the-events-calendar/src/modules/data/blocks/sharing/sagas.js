/**
 * External Dependencies
 */
import { put, all, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';

export function* setInitialState( action ) {
	const { get } = action.payload;

	yield all( [
		put(
			actions.setGoogleCalendarLabel(
				get( 'googleCalendarLabel', DEFAULT_STATE.googleCalendarLabel )
			)
		),
		put( actions.setiCalLabel( get( 'iCalLabel', DEFAULT_STATE.iCalLabel ) ) ),
		put( actions.setHasIcal( get( 'hasiCal', DEFAULT_STATE.hasiCal ) ) ),
		put(
			actions.setHasGoogleCalendar( get( 'hasGoogleCalendar', DEFAULT_STATE.hasGoogleCalendar ) )
		),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
