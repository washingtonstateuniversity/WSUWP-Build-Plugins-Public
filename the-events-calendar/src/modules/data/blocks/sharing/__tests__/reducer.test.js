/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/sharing';
import reducer, { DEFAULT_STATE } from '@moderntribe/events/data/blocks/sharing/reducer';

describe( '[STORE] - Sharing reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set the iCal Label', () => {
		expect( reducer( DEFAULT_STATE, actions.setiCalLabel( 'Modern Tribe iCal' ) ) )
			.toMatchSnapshot();
	} );

	it( 'Should set the google calendar label', () => {
		const expected = reducer(
			DEFAULT_STATE,
			actions.setGoogleCalendarLabel( 'Modern Tribe Google Calendar' )
		);
		expect( expected ).toMatchSnapshot();
	} );

	it( 'Should set if has the google calendar label', () => {
		expect( reducer( DEFAULT_STATE, actions.setHasGoogleCalendar( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setHasGoogleCalendar( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should set if has the iCal Label', () => {
		expect( reducer( DEFAULT_STATE, actions.setHasIcal( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setHasIcal( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should toggle if has the iCal Label', () => {
		expect( reducer( DEFAULT_STATE, actions.toggleIcalLabel() ) ).toMatchSnapshot();
	} );

	it( 'Should toggle if has the google label', () => {
		expect( reducer( DEFAULT_STATE, actions.toggleGoogleCalendar() ) ).toMatchSnapshot();
	} );
} );
