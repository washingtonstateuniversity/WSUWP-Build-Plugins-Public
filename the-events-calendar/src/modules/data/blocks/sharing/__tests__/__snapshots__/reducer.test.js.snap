// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Sharing reducer Should set if has the google calendar label 1`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": true,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should set if has the google calendar label 2`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": false,
  "hasiCal": true,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should set if has the iCal Label 1`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": true,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should set if has the iCal Label 2`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": false,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should set the google calendar label 1`] = `
Object {
  "googleCalendarLabel": "Modern Tribe Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": true,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should set the iCal Label 1`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": true,
  "iCalLabel": "Modern Tribe iCal",
}
`;

exports[`[STORE] - Sharing reducer Should toggle if has the google label 1`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": false,
  "hasiCal": true,
  "iCalLabel": "iCal Export",
}
`;

exports[`[STORE] - Sharing reducer Should toggle if has the iCal Label 1`] = `
Object {
  "googleCalendarLabel": "Google Calendar",
  "hasGoogleCalendar": true,
  "hasiCal": false,
  "iCalLabel": "iCal Export",
}
`;
